src/util/sss_sockets.c

```diff
@@ -97,6 +98,27 @@ static errno_t set_fd_common_opts(int fd)
                   strerror(ret));
     }
 
+    if (timeout > 0) {
+        /* Set socket read & write timeout */
+        tv = tevent_timeval_current_ofs(timeout, 0);
```

Comment:
If I understand correctly, `tevent_timeval_current_ofs` returns `time(NULL) + ofs`:
```
struct timeval tevent_timeval_current_ofs(uint32_t secs, uint32_t usecs)
{
        struct timeval tv = tevent_timeval_current();
        return tevent_timeval_add(&tv, secs, usecs);
}
```
  --  that's used to schedule tevents with `offset` in the future.

I think, this is not appropriate for `SO_RCVTIMEO` and `SO_SNDTIMEO`: `If an  input  or  output function  blocks  for  this  period  of  time`

Reply:
Thank you, my copy/paste mistake. Of course tevent_timeval_set must be used here.